firewood/src/merkle/node/extension.rs

```diff
@@ -127,12 +134,88 @@ impl Storable for ExtNode {
     }
 
     fn deserialize<T: crate::shale::CachedStore>(
-        _addr: usize,
-        _mem: &T,
+        mut offset: usize,
+        mem: &T,
     ) -> Result<Self, crate::shale::ShaleError>
     where
         Self: Sized,
     {
-        todo!()
+        let header_size = Self::PATH_LEN_SIZE + DiskAddress::MSIZE;
+
+        let path_and_disk_address = mem
+            .get_view(offset, header_size)
+            .ok_or(InvalidCacheView {
+                offset,
+                size: header_size,
+            })?
+            .as_deref();
+
+        offset += header_size as usize;
+
+        let mut cursor = Cursor::new(path_and_disk_address);
+        let mut buf = [0u8; DiskAddress::MSIZE as usize];
+
+        let path_len = {
+            let buf = &mut buf[..Self::PATH_LEN_SIZE as usize];
+            cursor.read_exact(buf)?;
+            buf[0] as u64
+        };
+
+        let disk_address = {
+            cursor.read_exact(buf.as_mut())?;
+            DiskAddress::from(u64::from_le_bytes(buf) as usize)
+        };
+
+        let path = mem
+            .get_view(offset, path_len)
+            .ok_or(InvalidCacheView {
+                offset,
+                size: path_len,
+            })?
+            .as_deref();
+
+        offset += path_len as usize;
+
+        let path: Vec<u8> = path.into_iter().flat_map(to_nibble_array).collect();
+
+        let path = PartialPath::decode(&path).0;
+
+        let encoded_len_raw = mem
+            .get_view(offset, Self::DATA_LEN_SIZE)
+            .ok_or(InvalidCacheView {
+                offset,
+                size: Self::DATA_LEN_SIZE,
+            })?
+            .as_deref();
+
+        offset += Self::DATA_LEN_SIZE as usize;
+
+        let mut cursor = Cursor::new(encoded_len_raw);
+
+        let encoded_len = {
+            let mut buf = [0u8; Self::DATA_LEN_SIZE as usize];
+            cursor.read_exact(buf.as_mut())?;
+            DataLen::from_le_bytes(buf) as u64
+        };
+
+        let encoded = if encoded_len != 0 {
+            let encoded = mem
+                .get_view(offset, encoded_len)
+                .ok_or(InvalidCacheView {
+                    offset,
+                    size: encoded_len,
+                })?
+                .as_deref();
+
+            encoded.into()
```

Comment:
nit: hard to read; too many encoded :)
```suggestion
            Some(
                mem.get_view(offset, encoded_len)
                    .ok_or(InvalidCacheView {
                        offset,
                        size: encoded_len,
                    })?
                    .as_deref(),
            )
```